src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresViewIT.java

```diff
@@ -87,69 +91,131 @@ public void after() throws SQLException {
-          assertNull(rs.getString("lastname"));
-        }
-      }
-    }
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(topic, schema, struct));
+    connect.kafka().produce(topic, null, kafkaValue);
+
+    waitForCondition(
+        () -> {
+          try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
+            try (Statement s = c.createStatement()) {
+              try (ResultSet rs = s.executeQuery("SELECT * FROM " + topic)) {
+                boolean result = rs.next()
+                    && struct.getString("firstname").equals(rs.getString("firstname"))
+                    && rs.getString("lastname") == null;
+                return Optional.of(result).orElse(false);
+              }
+            }
+          }
+        },
+        VERIFY_MAX_DURATION_MS,
+        "The database content did not match the record's content."
+    );
   }
 
   @Test
-  public void testWriteToView() throws SQLException {
+  public void testWriteToView() throws Exception {
     createTestTableAndView("firstName, lastName");
-    startTask();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
     final Schema schema = SchemaBuilder.struct().name("com.example.Person")
         .field("firstname", Schema.STRING_SCHEMA)
         .field("lastname", Schema.STRING_SCHEMA)
         .build();
     final Struct struct = new Struct(schema)
         .put("firstname", "Christina")
         .put("lastname", "Brams");
-    task.put(Collections.singleton(new SinkRecord(topic, 1, null, null, schema, struct, 1)));
-    try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
-      try (Statement s = c.createStatement()) {
-        try (ResultSet rs = s.executeQuery("SELECT * FROM " + topic)) {
-          assertTrue(rs.next());
-          assertEquals(struct.getString("firstname"), rs.getString("firstname"));
-          assertEquals(struct.getString("lastname"), rs.getString("lastname"));
-        }
-      }
-    }
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(topic, schema, struct));
+    connect.kafka().produce(topic, null, kafkaValue);
+
+    waitForCondition(
+        () -> {
+          try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
+            try (Statement s = c.createStatement()) {
+              try (ResultSet rs = s.executeQuery("SELECT * FROM " + topic)) {
+                boolean result = rs.next()
+                    && struct.getString("firstname").equals(rs.getString("firstname"))
+                    && struct.getString("lastname").equals(rs.getString("lastname"));
+                return Optional.of(result).orElse(false);
+              }
+            }
+          }
+        },
+        VERIFY_MAX_DURATION_MS,
+        "The database content did not match the record's content."
+    );
```

Comment:
Same comment here w/r/t waiting for offsets and then performing a single database content assertion.